src/som/interpreter/nodes/ExpressionWithReceiverNode.java

```diff
@@ -15,6 +15,8 @@ public ExpressionWithReceiverNode(final SourceSection sourceSection) {
   public Object evaluateReceiver(final VirtualFrame frame){
     return this.getReceiver().executeGeneric(frame);
   }
+
+  public abstract Object executeGenericWithReceiver(final VirtualFrame frame, final Object receiver);
```

Comment:
I think I will incorporate this version to Mate. I agree it would be a great to make a performance comparison. However, at a first sight I do not see a fundamental difference on that sense. Yes in the sense that this version is not transparent to the previous message send implementation. 
Dardo could you separate both solutions in two different branches so we eventually benchmark both approaches?